### DISCUSSION TITLE
Top-Level Inheritdoc For Tags

### DISCUSSION THREAD

**Discussion by colsonab** [EWSoftware/SHFB#993] [General]
I am using the Default2022 style.

I notice that _top-level_ inheritance of constructs in XML code documentation works with intellisense within Visual Studio but doesn't seem to work in terms of getting processed the same way when generating documentation.  Consider the following:

```
namespace MyNamespace
{
    /// <summary>
    /// <para>
    /// A new <i>sample</i> summary.
    /// </para>
    /// </summary>
    /// <inheritdoc cref="MyMethod" path="/remarks"/>
    public class MyClass
    {
        /// <summary>
        /// Registers services for dependency injection.
        /// </summary>
        /// <remarks>
        /// <para>
        /// These are remarks about MyMethod()!
        /// </para>
        /// </remarks>
        public void MyMethod()
        {
        }
    }
}
```

The compiled .xml documentation file looks like (with &lt;inheritdoc&gt; tags):

```
<?xml version="1.0"?>
<doc>
    <assembly>
        <name>ShfbBug</name>
    </assembly>
    <members>
        <member name="T:MyNamespace.MyClass">
            <summary>
            <para>
            A new <i>sample</i> summary.
            </para>
            </summary>
            <inheritdoc cref="M:MyNamespace.MyClass.MyMethod" path="/remarks"/>
        </member>
        <member name="M:MyNamespace.MyClass.MyMethod">
            <summary>
            Registers services for dependency injection.
            </summary>
            <remarks>
            <para>
            These are remarks about MyMethod()!
            </para>
            </remarks>
        </member>
    </members>
</doc>
```
The temporary SHFB file _InheritedDocs_.xml file looks like:
```
<doc>
  <assembly>
    <name>_InheritedDocs_</name>
  </assembly>
  <members>
  <member name="T:MyNamespace.MyClass">
  <summary>
    <para>
            A new <i>sample</i> summary.
            </para>
  </summary>
  
</member></members>
</doc>
```
The class does not inherit the method's remarks.

It is interesting to note that if the top-level tag

    /// <inheritdoc cref="MyMethod" path="/remarks"/>

gets nested and surrounded by a standard top-level tag 

    /// <remarks>
    /// <inheritdoc cref="MyMethod" path="/remarks/*"/>
    /// </remarks>

that the remarks get inherited as expected.

So, the question is how to get the new Default2022 style to process standard (or custom) top-level &lt;inheritdoc&gt; tags?  The differences above with inheriting remarks is probably cosmetic in most cases as it would be essentially equal to do just the 2nd nesting option but there are ramifications when trying to apply the 1st option to tags like &lt;param&gt; which are top-level tags and not nested.

**Comment by EWSoftware** [comment]
Remove the leading slash on the path when at the root level.  It's likely a matter of evaluation context though I'd have to test it to be sure.
